### PR TITLE
First draft of test generation c-script

### PR DESCRIPTION
**Pull request by Hector-Cardenas**
Wrote C script to generate testcases quickly. Currently is a standalone file without includes and defines but can consider bringing those in if needed. 

Test file format is all integers, in the following format:

`<command> <set start> <set end> <tag start> <tag end> <hit code>`

TestGen will then iterate through the set indexes and tag indexes, concatenating both with the hit code into a single address before printing on a new line with the given command.

### CODE REVIEW

**Review by stweeks-pdx — APPROVED** [review]
LGTM so far, might want to add a makefile runner for this

File: testing/GenFiles/TestGen.c [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+#include <stdlib.h>
+#include <stdbool.h>
+
+#define FILEBUFFS 20
+#define BYTESELECTWIDTH 6
+#define INDEXWIDTH	14
+#define TAGWIDTH	12
+
+int main(int argc, char *argv[]) {
+	if(argc < 3) printf("TestGen <input file> <output file>\n");
```
Comment (stweeks-pdx) [inline testing/GenFiles/TestGen.c]: This should exit if we don't give it 3 argc, right now it doesn't.

Comment (Hector-Cardenas) [reply testing/GenFiles/TestGen.c]: Woops. I meant for it to but then just forgot to type it

File: src/Makefile [hunk 2]
```diff
@@ -10,14 +10,17 @@ CC=gcc
...
-
+TGDIR=../testing/GenFiles/
 
 _DEPS = defines.h trace.h
 DEPS = $(patsubst %,$(IDIR)/%,$(_DEPS))
 
 _OBJ = main.o trace.o
 OBJ = $(patsubst %,$(ODIR)/%,$(_OBJ))
 
+_TEST = TestGen.o
```
Comment (Hector-Cardenas) [inline src/Makefile:21]: Uh... Genuinely not 100% on anything I added here; just tried to imitate what was going on with the others. Please correct as needed.

**Comment by Hector-Cardenas** [comment]
New push now uses n=7 as an internal NOP that allows us to add comments to trace files; parser has been updated to handle this better. In-line comments should still be viable, but this gives us more control when writing testgen files. Removed some redundant defines, and attempted to do something with Makefile... (It works?).

File: src/Makefile (hunk 2 above)
Comment (stweeks-pdx) [reply src/Makefile:21]: Yes this is correct

**Review by stweeks-pdx — APPROVED** [review]

**Pull request merged**